src/R/Core/Impl/AST/AstNode.cs

```diff
@@ -124,10 +124,10 @@ public abstract class AstNode : IAstNode {
         /// <summary>
         /// Finds deepest node that fully encloses given range
         /// </summary>
-        public virtual IAstNode NodeFromRange(ITextRange range) {
+        public virtual IAstNode NodeFromRange(ITextRange range, bool endInclusive = false) {
             IAstNode node = null;
 
-            if (TextRange.Contains(this, range)) {
+            if (endInclusive ? TextRange.ContainsInclusiveEnd(this, range) : TextRange.Contains(this, range)) {
```

Comment:
Sure. Weird that I didn't spot it. I'll rename the parameter to `inclusiveEnd` as well, so that it matches.